tests/integration.rs

```diff
@@ -1024,6 +1024,43 @@ mod tests {
         handle_child_output(r, &output);
     }
 
+    #[allow(unused_variables)]
+    fn _test_power_button(acpi: bool) {
+        let focal = UbuntuDiskConfig::new(FOCAL_IMAGE_NAME.to_string());
+        let guest = Guest::new(Box::new(focal));
+        let mut cmd = GuestCommand::new(&guest);
+        let api_socket = temp_api_path(&guest.tmp_dir);
+
+        #[cfg(target_arch = "x86_64")]
+        let kernel_path = direct_kernel_boot_path();
+        #[cfg(target_arch = "aarch64")]
+        let kernel_path = if acpi {
+            edk2_path()
+        } else {
+            direct_kernel_boot_path()
+        };
+
+        cmd.args(&["--cpus", "boot=1"])
+            .args(&["--memory", "size=512M"])
+            .args(&["--kernel", kernel_path.to_str().unwrap()])
+            .args(&["--cmdline", DIRECT_KERNEL_BOOT_CMDLINE])
+            .capture_output()
+            .default_disks()
+            .default_net()
+            .args(&["--api-socket", &api_socket]);
+
+        let child = cmd.spawn().unwrap();
+
+        let r = std::panic::catch_unwind(|| {
+            guest.wait_vm_boot(None).unwrap();
+            assert!(remote_command(&api_socket, "power-button", None));
+        });
+
+        let output = child.wait_with_output().unwrap();
```

Comment:
Nit: one potential improvement is using "wait_timeout" to avoid blindly waiting and report error promptly. Just for your reference, this is how it was done from "test_live_migration": https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/tests/integration.rs#L6921-L6932.

Reply:
@MrXinWang Will leave the above comments up to you and it can be addressed in a separate PR.

Reply:
Hi @likebreath, thank you. In this PR, I just extracted the old code to a common function and use the common function to test DT and ACPI. I will remember this comment and try to address it in future PR.